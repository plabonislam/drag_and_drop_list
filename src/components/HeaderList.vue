<script setup>
const headers = ['TO DO', 'IN PROGRESS', 'QA READY', 'QA', 'RELEASE READY', 'DONE']
</script>

<template>
  <tr>
    <th v-for="(title, index) in headers" :key="index">{{ title }}</th>
  </tr>
</template>
