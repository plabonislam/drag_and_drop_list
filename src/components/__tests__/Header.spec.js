import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import HeaderList from "../HeaderList.vue"

describe('Table Header', () => {
    it('renders properly', async () => {
        const wrapper = mount(HeaderList)
        const list = wrapper.vm.headers;
        wrapper.vm.headers = [...list, "to mo"]
        expect(wrapper.html()).toMatchSnapshot();
    })

    it('setValue for input field', async () => {
        const wrapper = mount(HeaderList)

        expect(wrapper.vm.t1).toEqual('')
        expect(wrapper.vm.t2).toEqual('')

        // for single setValue
        wrapper.find('.foo').setValue('single-foo');
        expect(wrapper.vm.t1).toEqual('single-foo')

        // for array
        const wrapperArray = wrapper.findAll('.foo')
        wrapperArray.forEach(wrapper => wrapper.setValue('foo'))
        expect(wrapper.vm.t1).toEqual('foo')
        expect(wrapper.vm.t2).toEqual('foo')

        // directly setValue for vm
        wrapper.vm.t1 = 'test'
        expect(wrapper.vm.t1).toEqual('test')
    })

    it('adds a new header', async () => {
        const wrapper = mount(HeaderList)
        const initialLength = wrapper.vm.headers.length
        wrapper.vm.headers.push('New Header')
        expect(wrapper.vm.headers.length).toBe(initialLength + 1)
        expect(wrapper.vm.headers).toContain('New Header')
    })

    it('removes a header', async () => {
        const wrapper = mount(HeaderList)
        const initialLength = wrapper.vm.headers.length
        wrapper.vm.headers.pop()
        expect(wrapper.vm.headers.length).toBe(initialLength - 1)
    })

    it('checks if header list is empty', async () => {
        const wrapper = mount(HeaderList)
        wrapper.vm.headers = []
        expect(wrapper.vm.headers.length).toBe(0)
    })

    it('checks header list length', async () => {
        const wrapper = mount(HeaderList)
        const length = wrapper.vm.headers.length
        expect(wrapper.vm.headers.length).toBe(length)
    })

    it('checks if a specific header exists', async () => {
        const wrapper = mount(HeaderList)
        const header = 'Specific Header'
        wrapper.vm.headers.push(header)
        expect(wrapper.vm.headers).toContain(header)
    })
})