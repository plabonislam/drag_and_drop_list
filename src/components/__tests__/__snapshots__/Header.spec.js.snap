// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table Header > renders properly 1`] = `
"<div><input type="text" name="t1" class="foo"><input type="text" name="t2" class="foo"></div>
<tr>
  <th>TO DO</th>
  <th>IN PROGRESS</th>
  <th>QA READY</th>
  <th>QA</th>
  <th>RELEASE READY</th>
  <th>DONE</th>
</tr>"
`;
